src/main/java/edu/greenblitz/tobyDetermined/subsystems/telescopicArm/Extender.java

```diff
@@ -0,0 +1,123 @@
+package edu.greenblitz.tobyDetermined.subsystems.telescopicArm;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel;
+import edu.greenblitz.tobyDetermined.RobotMap;
+import edu.greenblitz.tobyDetermined.subsystems.GBSubsystem;
+import edu.greenblitz.utils.PIDObject;
+import edu.greenblitz.utils.motors.GBSparkMax;
+import edu.wpi.first.math.util.Units;
+
+public class Extender extends GBSubsystem {
+    public static final int EXTENDER_MOTOR_ID = 0;
+    public static final int BACKWARDS_LIMIT = 0;
+    public static final double FORWARD_LIMIT = 0.6;
+    public static final double distanceBetweenHoles = 6.35;
+    public static final double outputGearAmountOfTeeth = 32;
+    public static final double gearRatio = 1 / 7.0;
+
+    public static final double maxLengthInRobot = 0.4;
+    public static final PIDObject extenderPID = new PIDObject();
+    public static final double extenderConversionFactor =
+            (((RobotMap.General.Motors.SPARKMAX_TICKS_PER_RADIAN / gearRatio) * outputGearAmountOfTeeth) / (2 * Math.PI) ) * distanceBetweenHoles;
+
+    private static ExtenderState state = ExtenderState.CLOSED;
+
+
+    private static Extender instance;
+    private GBSparkMax motor;
+
+    public static Extender getInstance (){
+        if(instance == null){
+            instance = new Extender();
+        }
+        return instance;
+    }
+
+    private Extender(){
+        motor = new GBSparkMax(EXTENDER_MOTOR_ID, CANSparkMaxLowLevel.MotorType.kBrushless);
+        motor.getEncoder().setPosition(0); //maybe absolute encoder+*
+        motor.config(new GBSparkMax.SparkMaxConfObject()
+                .withPID(extenderPID)
+                .withPositionConversionFactor(extenderConversionFactor)
+        );
+        motor.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, BACKWARDS_LIMIT);
+        motor.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, (float) FORWARD_LIMIT);
```

Comment:
I know that the function needs a float, but it is called inside a GB wrapper, so maybe we can convert the variable type inside that function?